src/PackageSelection/PackageSelection.php

```diff
@@ -461,4 +491,26 @@ private function filterRepositories(array $repositories)
             return true;
         });
     }
+
+    /**
+     * @param Pool $pool
+     * @param $name
+     * @param Link $link
+     * @return array|\Composer\Package\PackageInterface[]
+     */
+    private function findMatches(Pool $pool, $name, Link $link)
+    {
+        try {
+            if (!$this->bestCandidateStrategy) {
+                return $pool->whatProvides($name, $link->getConstraint(), true);
+            } else {
+                $selector = new VersionSelector($pool);
+                if ($best = $selector->findBestCandidate($name, str_replace(['[', ']'], ['', ''], $link->getConstraint()->getPrettyString()))) {
+                    return [$best];
+                }
+            }
+        } catch (\Exception $e) {
```

Comment:
well, this is a separate topic than restricting versions IMO (unless this happens only in the VersionSelector, in which case the try/catch should wrap only this part of the code)